src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -161,11 +183,15 @@ private void configureMultiSinkWithSchema(BatchSinkContext context,
       }
       String name = key.substring(TABLE_PREFIX.length());
       Schema schema = Schema.parseJson(argument.getValue());
+
       // TODO: (CDAP-14600) pass in schema as an argument instead of using macros and setting arguments
       // add better platform support to allow passing in arguments when instantiating a plugin
       context.getArguments().set(SCHEMA_MACRO, schema.toString());
-      ValidatingOutputFormat validatingOutputFormat = context.newPluginInstance(FORMAT_PLUGIN_ID);
 
+      String macroFormat = this.config.format;
```

Comment:
use `macroFormat = this.config.getFormatName();`

Reply:
removed this piece of code, verified that its not needed, with the fact that format has been resolved at this stage